billing-db/sprocs/update_resources.sql

```diff
@@ -0,0 +1,428 @@
+-- Do not run too far back in time or will take too long to update resources during which time users won't be able to generate billing reports
```

Comment:
I don't feel very qualified to review the SQL here, because 1. I'm far from a SQL expert, and 2. I'm not familiar with plpgsql specifically.

However, reading the comments, I will say that it sounds like maybe we need some little utilities for manipulating the usage events, so that everything is done properly.